### DISCUSSION TITLE
read from yaml config file some struct item is null

### DISCUSSION THREAD

**Discussion by xiangxud** [spf13/viper#1067] [Q&A] [answered]
type ConfYaml struct {
	Logger   logger.Config            `yaml:"logger"`
	Identity identity.IdentitySection `yaml:"identity"`
	IP       identity.IPSection       `yaml:"ip"`
	Stra     stra.StraSection         `yaml:"stra"`
	Enable   enableSection            `yaml:"enable"`
	Report   reportSection            `yaml:"report"`
	Sys      sys.SysSection           `yaml:"sys"`
	UI       uiSection                `yaml:"ui"`
	Mqtt     mqttSection              `yaml:"mqtt"`
	Screen   screenSection            `yaml:"screen"`
}
type screenSection struct {
	BUsed             bool   `yaml:"bused"`
	CaptureTimeDelay  uint   `yaml:"capturetimedelay"`
	JpegFileLocalPath string `yaml:"jpeglocalpath"`
	JpegFileURL       string `yaml:"jpegurl"`
}

type mqttSection struct {
	SUBTOPIC      string `yaml:"subtopic"` //"topic1"
	PUBTOPIC      string `yaml:"pubtopic"`
	QOS           byte   `yaml:"qos"`           //1
	SERVERADDRESS string `yaml:"serveraddress"` //= "tcp://mosquitto:1883"
	CLIENTID      string `yaml:"clientid"`      //= "mqtt_subscriber"
	WRITETOLOG    bool   `yaml:"writelog"`      //= true  // If true then received messages will be written to the console
	WRITETODISK   bool   `yaml:"writetodisk"`   //= false // If true then received messages will be written to the file below
	OUTPUTFILE    string `yaml:"outputfile"`    //= "/binds/receivedMessages.txt"
	//	CommandLocalPath string `yam:"commanlocalpath"`
}
my yaml file 
mqtt:
  subtopic: "Control/#"
  pubtopic: "bbt"
  qos: 1
  serveraddress: "tcp://10.0.23.123:1883"
  clientid: "mqtt_subscriber"
  writelog: true
  writetodisk: true
  outputfile: "./receivedMessages.txt"
screen:
  bused: true
  capturetimedelay: 600
  jpeglocalpath: "jpg/"                                  ??? why is null
  jpegurl: "ftp://192.168.0.25/file"                ??? why is null

i use like this:
	err = viper.Unmarshal(&Config)
but 
	BUsed             ok  is true
	CaptureTimeDelay   ok  is 600
	JpegFileLocalPath string  is  ""  must is  "jpg/"   why ？？？？
	JpegFileURL   is  ""    must is  "ftp://192.168.0.25/file"   why  ???

  WRITETOLOG is false          must is true why ???

**Comment by sagikazarmark** [comment, marked as answer]
Take a look at https://github.com/spf13/viper#unmarshaling-doesnt-work

**Comment by xiangxud** [comment]
I change my struct maping(add mapstructure key words)  like this:
	UI       uiSection                `yaml:"ui" mapstructure:"ui"`
	Mqtt     mqttSection              `yaml:"mqtt" mapstructure:"mqtt"`
	Screen   screenSection            `yaml:"screen" mapstructure:"screen"`

but 
 come to the same thing
Maybe I misunderstand

**Comment by xiangxud** [comment]
thanks ，i add  mapstructure key word in struct item ，it is ok
